libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -132,6 +130,33 @@ static _mqttConnection_t * _createMqttConnection( bool awsIotMqttMode,
  */
 static void _destroyMqttConnection( _mqttConnection_t * pMqttConnection );
 
+/**
+ * @brief Common setup function for subscribe & unsubscribe operations.
+ *
+ * See @ref mqtt_function_subscribeasync or @ref mqtt_function_unsubscribeasync for a
+ * description of the parameters and return values.
+ */
+static IotMqttError_t _subscriptionCommonSetup( IotMqttOperationType_t operation,
+                                          IotMqttConnection_t mqttConnection,
+                                          const IotMqttSubscription_t * pSubscriptionList,
+                                          size_t subscriptionCount,
+                                          uint32_t flags,
+                                          IotMqttOperation_t * const pOperationReference );
+/**
+ * @brief Utility function for creating & serializing subscribe/unsubscribe requests
+ *
+ * See @ref mqtt_function_subscribeasync or @ref mqtt_function_unsubscribeasync for a
+ * description of the parameters and return values.
+ */
+static IotMqttError_t _subscriptionCreateAndSerialize( IotMqttOperationType_t operation,
```

Comment:
These 3 functions have an excessive number of parameters; could this be reduced somehow?

Reply:
`_subscriptionCommon()` already had [7 arguments before this change](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/standard/mqtt/src/iot_mqtt_api.c#L142), now it has 8 and the 2 new helpers have either that number or less.  Reducing the number of arguments further will require much riskier refactoring, and the purpose of this change is to improve readability with minimum risk.